### PR TITLE
Network: Properly handle OVN uplink with disabled DHCP/DHCPv6 (from Incus)

### PR DESCRIPTION
**Pull request by kadinsayani**
(cherry picked from https://github.com/lxc/incus/pull/343)

### CODE REVIEW

File: lxd/network/driver_ovn.go [hunk 1]
```diff
@@ -1245,7 +1245,16 @@ func (n *ovn) allocateUplinkPortIPs(uplinkNet Network, routerMAC net.HardwareAdd
 					return fmt.Errorf(`Missing required "ipv4.ovn.ranges" config key on uplink network`)
 				}
 
-				ipRanges, err := shared.ParseIPRanges(uplinkNetConf["ipv4.ovn.ranges"], uplinkNet.DHCPv4Subnet())
+				dhcpSubnet := uplinkNet.DHCPv4Subnet()
+				allowedNets := []*net.IPNet{}
```
Comment (MggMuggins) [inline lxd/network/driver_ovn.go:1249]: Optional: Would it be possible to dispense with the array and just use a *net.IPNet instead? Could default to uplinkIPv4Net and change the pointer if dhcpSubnet isn't nil.

**Review by MggMuggins — COMMENTED** [review]
Just one nit.

I assume this makes it possible to use a managed bridge with DHCP disabled as an UPLINK, are there any other use-cases this enables?

**Comment by kadinsayani** [comment]
> 
> 
> 
> 
> Just one nit.
> 
> 
> 
> I assume this makes it possible to use a managed bridge with DHCP disabled as an UPLINK, are there any other use-cases this enables?

You got it, we want to properly handle 'lxc network create <name> ipv4.dhcp=false ipv6.dhcp=false'. There aren't any other use cases that I'm aware of.

File: lxd/network/driver_ovn.go (hunk 1 above)
Comment (kadinsayani) [reply lxd/network/driver_ovn.go:1249]: I think we should keep it as a slice in case IPv6 and IPv4 OVN ranges are specified.

**Review by tomponline — APPROVED** [review]

**Pull request merged**